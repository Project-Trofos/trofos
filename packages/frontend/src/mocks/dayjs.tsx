const dayjsMock = {
  year: () => '2022',
};

export default dayjsMock;
