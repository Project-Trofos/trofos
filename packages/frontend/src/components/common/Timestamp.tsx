import React from 'react';
import { Tag } from 'antd';
import dayjs, { Dayjs } from 'dayjs';

export type TimestampProps = {
  createdAt: string | Dayjs;
  updatedAt: string | Dayjs | undefined;
};

export default function Timestamp(props: TimestampProps) {
  const { createdAt, updatedAt } = props;

  const createdAtDayjs: Dayjs = typeof createdAt === 'string' ? dayjs(createdAt) : createdAt;
  const updatedAtDayjs: Dayjs | undefined = typeof updatedAt === 'string' ? dayjs(updatedAt) : updatedAt;

  return (
    <div>
      {updatedAtDayjs ? (
        <div>
          {updatedAtDayjs.format('DD/MM/YYYY HH:mm')} <Tag>Edited</Tag>
        </div>
      ) : (
        <div>{createdAtDayjs.format('DD/MM/YYYY HH:mm')}</div>
      )}
    </div>
  );
}
