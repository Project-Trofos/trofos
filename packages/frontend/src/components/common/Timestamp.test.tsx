import React from 'react';
import dayjs from 'dayjs';
import { render, screen } from '@testing-library/react';
import Timestamp from './Timestamp';

describe('test Timestamp', () => {
  describe('when rendering', () => {
    it('should render normal timestamp correctly', () => {
      const { baseElement } = render(<Timestamp createdAt={dayjs('2023-01-01')} updatedAt={undefined} />);

      expect(screen.getByText('01/01/2023', { exact: false })).toBeInTheDocument();

      expect(baseElement).toMatchSnapshot();
    });

    it('should render edited timestamp correctly', () => {
      const { baseElement } = render(<Timestamp createdAt={dayjs('2023-01-01')} updatedAt={dayjs('2023-01-02')} />);

      expect(screen.getByText('02/01/2023', { exact: false })).toBeInTheDocument();
      expect(screen.getByText(/edited/i)).toBeInTheDocument();

      expect(baseElement).toMatchSnapshot();
    });
  });
});
