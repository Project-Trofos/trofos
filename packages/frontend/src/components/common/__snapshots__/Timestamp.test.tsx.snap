// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Timestamp > when rendering > should render edited timestamp correctly 1`] = `
<body>
  <div>
    <div>
      <div>
        02/01/2023 00:00
         
        <span
          class="ant-tag"
        >
          Edited
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`test Timestamp > when rendering > should render normal timestamp correctly 1`] = `
<body>
  <div>
    <div>
      <div>
        01/01/2023 00:00
      </div>
    </div>
  </div>
</body>
`;
